src/Compatibility/GenAPI/Microsoft.DotNet.GenAPI/CSharpAssemblyDocumentGenerator.cs

```diff
@@ -0,0 +1,370 @@
+        if (member.IsOverride)
+        {
+            return false;
+        }
+
+        if (member.ContainingType.BaseType is not INamedTypeSymbol baseType)
+        {
+            return false;
+        }
+
+        if (member is IMethodSymbol method)
+        {
+            if (method.MethodKind == MethodKind.ExplicitInterfaceImplementation)
+            {
+                return false;
+            }
+
+            // If they're methods, compare their names and signatures.
+            return baseType.GetMembers(member.Name)
+                .Any(baseMember => _symbolFilter.Include(baseMember) &&
+                     (baseMember.Kind != SymbolKind.Method ||
+                      method.SignatureEquals((IMethodSymbol)baseMember)));
+        }
+        else if (member is IPropertySymbol prop && prop.IsIndexer)
+        {
+            // If they're indexers, compare their signatures.
+            return baseType.GetMembers(member.Name)
+                .Any(baseMember => baseMember is IPropertySymbol baseProperty &&
+                     _symbolFilter.Include(baseMember) &&
+                     (prop.GetMethod.SignatureEquals(baseProperty.GetMethod) ||
+                      prop.SetMethod.SignatureEquals(baseProperty.SetMethod)));
+        }
+        else
+        {
+            // For all other kinds of members, compare their names.
+            return baseType.GetMembers(member.Name)
+                .Any(_symbolFilter.Include);
+        }
+    }
+
+    private SyntaxNode Visit(SyntaxNode namedTypeNode, INamedTypeSymbol namedType)
+    {
+        IEnumerable<ISymbol> members = namedType.GetMembers().Where(_symbolFilter.Include);
+
+        // If it's a value type
+        if (namedType.TypeKind == TypeKind.Struct)
+        {
+            namedTypeNode = _syntaxGenerator.AddMembers(namedTypeNode, namedType.SynthesizeDummyFields(_symbolFilter, _attributeDataSymbolFilter));
+        }
+
+        namedTypeNode = _syntaxGenerator.AddMembers(namedTypeNode, namedType.TryGetInternalDefaultConstructor(_symbolFilter));
+
+        foreach (ISymbol member in members.Order())
+        {
+            if (member is IMethodSymbol method)
+            {
+                // If the method is ExplicitInterfaceImplementation and is derived from an interface that was filtered out, we must filter it out as well.
+                if (method.MethodKind == MethodKind.ExplicitInterfaceImplementation &&
+                    method.ExplicitInterfaceImplementations.Any(m => !_symbolFilter.Include(m.ContainingSymbol) ||
+                    // if explicit interface implementation method has inaccessible type argument
+                    m.ContainingType.HasInaccessibleTypeArgument(_symbolFilter)))
+                {
+                    continue;
+                }
+
+                // Filter out default constructors since these will be added automatically
+                if (_hideImplicitDefaultConstructors && method.IsImplicitDefaultConstructor(_symbolFilter))
```

Comment:
This if is one of the few parts of the code in this file that I did modify. This new class takes a `hideImplicitDefaultConstructors` boolean. The previous class did not (see `CSharpFileBuilder.cs` line 201, left side).